src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs

```diff
@@ -53,10 +98,96 @@ private static bool NeedVtableAttribute(SyntaxNode node)
                 !GeneratorHelper.IsWinRTType(declaration); // Making sure it isn't an RCW we are projecting.
         }
 
-        private static VtableAttribute GetVtableAttributeToAdd(GeneratorSyntaxContext context)
+        // Filters to non WinRT types which are public and can be instantiated.
+        // This is used to try to determine types which a component source generator will process.
+        private static bool IsComponentType(SyntaxNode node)
+        {
+            return node is ClassDeclarationSyntax declaration &&
+                !declaration.Modifiers.Any(m => m.IsKind(SyntaxKind.StaticKeyword) || m.IsKind(SyntaxKind.AbstractKeyword)) &&
+                declaration.Modifiers.Any(m => m.IsKind(SyntaxKind.PublicKeyword)) &&
+                !GeneratorHelper.IsWinRTType(declaration); // Making sure it isn't an RCW we are projecting.
+        }
+
+        private static (VtableAttribute, List<VtableAttribute>) GetVtableAttributeToAdd(GeneratorSyntaxContext context)

```

Comment:
This is used in a transform, but `List<T>` isn't equatable. Should we return an `EquatableArray<T>` instead? I assume `VtableAttribute` is already equatable?

Reply:
Yea I started getting inconsistent with this, will address.

Reply:
Addressed.